app.py

```diff
@@ -0,0 +1,44 @@
+import os
+from config import Config
+from cdk_eoapi import pgStacInfra, vpc
+from aws_cdk import App
+import yaml
+
+app = App()
+
+try:
+    with open("config.yaml") as f:
+        config = yaml.safe_load(f)
+        config = (
+            {} if config is None else config
+        )  # if config is empty, set it to an empty dict
+        config = Config(**config)
+except FileNotFoundError:
+    # if no config at the expected path, using defaults
+    config = Config()
+
+vpc_stack = vpc.VpcStack(
+    scope=app,
+    id=config.build_service_name("pgSTAC-vpc"),
+    nat_gateway_count=config.nat_gateway_count,
+)
+
+
+pgstac_infra_stack = pgStacInfra.pgStacInfraStack(
```

Comment:
@emileten Let's add resource tagging for everything here with at least `project` using the project id.

Reply:
added this as a config option that defaults to the project id along with the stage if nothing is provided. 

Noting that if one passes values to `--tags` in the `cdk deploy` command, it overrides the config option.